src/testing/loadtest-azure/scripts/file-upload-to-loadtest.ps1

```diff
@@ -28,22 +34,50 @@ if (!(Test-Path $testFileName -PathType leaf)) {
   trow "File $testFileName does not exist"
 }
 
-$urlRoot = "https://" + $apiEndpoint + "/loadtests/" + $loadTestId + "/files/" + $testFileId
+$urlRoot = "https://{0}/loadtests/{1}/files/{2}"  -f $apiEndpoint, $loadTestId, $testFileId
 
 # Following is to get Invoke-RestMethod to work
-$url = $urlRoot + "?api-version=" + $apiVersion
+$url = "{0}?api-version={1}"  -f $urlRoot, $apiVersion
 
 Write-Verbose "*** Load test service data plane: $urlRoot"
 
 # Secure string to use access token with Invoke-RestMethod in Powershell
 $accessTokenSecure = ConvertTo-SecureString -String $accessToken -AsPlainText -Force
 
-Invoke-RestMethod `
+$result = Invoke-RestMethod `
   -Uri $url `
   -Method PUT `
   -Authentication Bearer `
   -Token $accessTokenSecure `
   -Form @{ file = Get-Item $testFileName } `
   -Verbose:$verbose -Debug
 
+# export pipeline variables
+if($pipeline) {
+  echo "##vso[task.setvariable variable=fileId]$($result.fileId)" # contains the fileId for in-pipeline usage
+} else {
+  $result
+}
+
+# wait till uploaded file is validated
+if($wait) {
+
+  do {
+
+    $fileStatus = (& $PSScriptRoot\loadtest-get-files.ps1 -apiEndpoint $apiEndpoint `
+                            -loadTestId $loadTestId `
+                            -fileId $($result.fileId) `
+                            -keepToken $true)
+    if ($fileStatus.validationStatus -ne "VALIDATION_SUCCESS") {
+      Write-Verbose "*** Waiting another 30s for file validation to complete $($fileStatus.validationStatus)"
+      Start-Sleep -seconds 30
```

Comment:
do you have any experience how long the validation usually takes? do we have to wait for 30 or is less also ok?

Reply:
I've no experience. It's new - there was no need to wait in the previous api version. As of now it looks like it takes only seconds (so most of the time it works w/o going through the loop) but sometimes it takes a little longer.

Reply:
Reduced it to 15 seconds